### PR TITLE
[IE] setBatchSize through reshape

### PR DESCRIPTION
**Pull request by jane-intel**

### CODE REVIEW

File: inference-engine/tests_deprecated/functional/shared_tests/network_tests/network_i8.hpp
Comment (ilya-lavrenov) [inline inference-engine/tests_deprecated/functional/shared_tests/network_tests/network_i8.hpp:347]: from my point of view, we need code like this:
```
        if (batch_size != 1) {
            if (network.getFunction()) {
                auto cnnNetworkImpl = std::make_shared<details::CNNNetworkImpl>(network);
                network = CNNNetwork(cnnNetworkImpl);
            } 
            network.setBatchSize(batch_size);
        }
```

File: inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp
Comment (ilyachur) [inline inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp:456]: nit: redundant line

**Review by ilyachur — COMMENTED** [review]
Can we add tests to check behaviour of setBatch/getBatch methods for nGraph function with dynamic shapes?

**Comment by jane-intel** [comment]
> Can we add tests to check behaviour of setBatch/getBatch methods for nGraph function with dynamic shapes?

Done

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
@jane-intel Please fix CI.

**Comment by jane-intel** [comment]
> @jane-intel Please fix CI.

There is a related MR to product-config -- pre-commit scope should be corrected? so I'm waiting for merge

**Comment by ilya-lavrenov** [comment]
Can we restore `setBatchSize` as is, but all changes move to `reshapeBatch` / `setBatchSizeReshape`? After, I think it can be merged

**Comment by jane-intel** [comment]
I suggest not to touch this methods in this release.

**Comment by jane-intel** [comment]
Closing in favour of https://github.com/openvinotoolkit/openvino/pull/2597 which is already merged

**Pull request closed without merging**
